From 784722a003157e66f7a6cad0292ee21de968a8fe Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Mon, 03 Feb 2020 15:57:05 +0200
Subject: [PATCH] [SW-8229] goya2: read efuse only once

To read the efuses takes a long time and should be done
only once. Define a static variable so the flag will be set
once we've finished reading the efuses.

Change-Id: Iec03810defd16f0353fa1d6bbb0eac8a5bde3d28
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/efuse.c b/common/efuse.c
index 8194ddb..412dac1 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -24,23 +24,25 @@
  */
 int fuse_read(u32 word, u32 *val)
 {
-	int i, offset;
+	int offset;
+	static uint16_t i = EFUSE_NOT_READ;
 
 	if (word > 64) {
 		return -1;
 	}
 
-	regmap_writel(1, mmPSOC_EFUSE_READ);
+	if (i <= EFUSE_NOT_READ)	{
+		regmap_writel(1, mmPSOC_EFUSE_READ);
 
-	for (i=0; i < FUSE_RETRIES; i++)
-		if (regmap_readl(mmPSOC_EFUSE_READ))
-			udelay(FUSE_DELAY_US);
-		else
-			break;
+		for (i = 0; i < FUSE_RETRIES; i++)
+			if (regmap_readl(mmPSOC_EFUSE_READ))
+				udelay(FUSE_DELAY_US);
+			else
+				break;
 
-	if (i == FUSE_RETRIES) {
-		return -2;
 	}
+	if (i == FUSE_RETRIES)
+		return -2;
 
 	/* read value from bank0 */
 	offset = (word * sizeof(u32));
diff --git a/common/efuse.h b/common/efuse.h
index 968b8ea..abdf77b 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -8,6 +8,10 @@
 #ifndef _EFUSE_H_
 #define _EFUSE_H_
 
+#define EFUSE_READ_DONE				0x2
+#define EFUSE_NOT_READ				0x1
+
+
 #define EFUSE_KEY_HASH_OFFSET			0x12
 #define EFUSE_SYSCFG_OFFSET			0x3A
 #define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
